lib/screens/profile/option_list.dart

```diff
@@ -0,0 +1,101 @@
+import 'package:flutter/material.dart';
+
+class OptionList extends StatelessWidget {
+  final List<Map<String, Object?>> _optionData;
+
+  const OptionList(this._optionData);
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      decoration: const BoxDecoration(
+        border: Border.symmetric(
+            horizontal: BorderSide(
+          width: 1,
+          color: Color.fromRGBO(204, 221, 255, 0.1),
+        )),
+        color: Color.fromRGBO(29, 40, 56, 1),
+      ),
+      child: ListView.separated(
+        shrinkWrap: true,
+        physics: const NeverScrollableScrollPhysics(),
+        itemCount: _optionData.length,
+        itemBuilder: (context, index) {
+          final optionName = _optionData[index]['optionName'] as String;
+          final currentSelection =
+              _optionData[index]['currentSelection'] as String?;
+          final onPressed = _optionData[index]['onPressed'] as VoidCallback;
+          return OptionButton(
+            optionName: optionName,
+            onPressed: onPressed,
+            currentSelection: currentSelection,
+          );
+        },
+        separatorBuilder: (context, index) {
+          return const Divider(
+            height: 1,
+            thickness: 1,
+            indent: 16,
+          );
+        },
+      ),
+    );
+  }
+}
+
+class OptionButton extends StatelessWidget {
```

Comment:
We should put public components in seperate files. But in this case it could be a private class (prefix it with _, like this: _OptionButton)